### ISSUE TITLE
tries: xml in cb_plugins:

### ISSUE THREAD

**Issue by jojo12** [jemproject/JEM-Project#942]
I tried with the old CB 1.9.1 and the new CB 2.0.8:
when I use: 
<?xml version="1.0" encoding="UTF-8"?>
<cbinstall version="1.0" type="plugin" group="user">
    <name>CB JEM MyEvents</name>
    <author>JEM Community</author>
    <creationDate>April 2015</creationDate>
    <version>2.0</version>
    <release>1.0.0</release>
it works well on both cb versions and is signed compatible with CB 2.0
(even when the language system is not the same, the old system works on CB 2.0

new here is the splitting of version and release!
to understand, you must edit my post!

**Comment by Hoffi1** [comment]
Thanks Jojo. After clicking "Edit" I could read your post understandable :wink: 
Will try it out so hopefully we could provide a single version for both CBs.

**Comment by jojo12** [comment]
sorry, I didn't look the preview. And even then in Github I don't knew how to post code. Now I will try. ;-)

**Comment by Hoffi1** [comment]
Hm... will try it with three back ticks:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<cbinstall version="1.0" type="plugin" group="user">
	<name>CB JEM MyEvents</name>
	<author>JEM Community</author>
	<creationDate>April 2015</creationDate>
	<version>2.0</version>
	<release>1.0.0</release>
```

Wow - seems to be perfect. Click Edit to see what I made. :smile:

**Comment by jojo12** [comment]
Wow wow, thank for this lesson!

**Comment by Hoffi1** [comment]
Taken the time to try it out - but it doesn't work.
If version is set to 2.0 it isn't installable on CB 1.9.1. It says "Plugin version (2.0) different from Community Builder version (1.9.1)"
But if version is set to 1.9 it can be installed on both CB 1.9.1 and 2.0.x - but on 2.0.x an excalation mark is shown with tooltip "This plugin is not compatible with your current CB version..."

So it seems there is no other way than providing different packages for CB 1.9 and 2.0. :cry:

**Comment by jojo12** [comment]
I don't understand:
I installed on a clean Joomla! 3.4.1 a CB 1.9.1 and installed our two cb plugins with the changes I mentioned. And it worked as easy like butter. Will try more.

**Comment by jojo12** [comment]
On a clean Joomla 3.4.1 with a new CB 1.9.1 I installed JEM 2.1.4, then I made the following tries:
1. I changed the cbfiles after installing them while comparing new plugins with old ones. ok
2. I made a new cbplugin zip and installed over the files from step 1. Ok
3. In uninstalled the plugin and installed this newly zipped plugin. All ok.

Can it be that the problem is only on Joomla 2.5.28 (CB 1.9.1, JEM 2.1.4) ?

**Comment by Hoffi1** [comment]
I would be really lucky if it would work. But on my CB 1.9.1 a plugin with version 2.0 was aborted with the message I wrote above. :cry: 
The only idea I can try is to set cbinstall version to 1.0 (which is currently 1.9 if I remember right). It would surprise me but will try it with hope.

**Comment by Hoffi1** [comment]
Our posts overlapped, thanks for your report.
I will made some more tries, also on J! 3.4 / CB 1.9.1.

**Comment by jojo12** [comment]
wait, perhaps I made the package with the old ones.

**Comment by Hoffi1** [comment]
I must wait because I'm @work - no chance to try yet. :wink:

**Comment by Hoffi1** [comment]
I checked exact your xml settings on J! 3.4 and then the code an would say it can't work:

(CB 1.9.1)```administrator/components/com_comprofiler/library/cd/cd.installer.php```  compares version tag against different strings between "1.9.1" and "1.0" - that's why "2.0" or something else starting with "2" isn't accepted.

(CB 2.0.9)```libraries/CBLib/CB/Legacy/cbInstallerPlugin.php``` have the same check but enhanced up to "2.0.9" so installer will accept old and new plugin versions.
But on ```libraries/CBLib/CB/Legacy/cbPluginHandler.php``` compatibility is checked by asking if the same version tag is at least 2.0 - no chance without quantum computer. :cry:

**Comment by jojo12** [comment]
I posted a question in the joomlapolis forum and got an answer. Can this help?

question:
Hi, there was earlier a 3rdparty plugin for eventlist. There is now the successor of eventlist JEM, the adapted plugins work fine on CB 1.9.1 and CB 2.0. There is only a problem of versioning. If we set in xml to version 1.9, in both cb versions the plugin is installed, but in cb 2.0 is a ! mark to say, it's not compatible.
If we set to version 2.0, in cb 1.91 it's not loaded with an error message.

Do we need really to create two identical plugins for this two cb versions?
Or do you have a proposal? Thank you in advance for a hin
answer:
The compatibly check is just checking that the XML version tag is at least 2.0.0. I guess you can just change the XML tag if you wanted, but that by no means makes the code suddenly upgraded. Depending on what you're needing to do most integration plugins have no purpose anymore as CB Auto Actions is capable of handling nearly any case.

**Comment by Hoffi1** [comment]
The first part of the answer is exactly what I read from CB's code / posted above. :wink: 
The "CB Auto Actions" part I don't understand. Seems there is something to learn...

**Comment by JKoelman** [comment]
> but that by no means makes the code suddenly upgraded

indeed it ain't as there will be a difference within the internal code of CB, but as we're not doing very complex things within the plugins it shouldn't matter much.

**Comment by Hoffi1** [comment]
Some time ago I tried to find out the differences between 1.9 and 2.0 user plugins.
On plugins delived and installed by CB package itself are indeed a lot of changes, mostly regarding language strings. But these string are stored in a single language file also delivered by CB - so this way can't be used by 3rd party plugins.
Then I compared the CB plugins delivered by the famous uddeIM package. And they are completely identical except the version number in the xml file.
Unfortunately there was no actual documentation available for CB 2.0 plugin developers.

Because I'm very busy with the big ACL thing top most on my list I would like to reduce other activities to it's minimum as good as possible. On the other side if I remember right there are more changes between pending plugins here (2.1.3) and their previous version provided on our Download page. So an intermediate release of these pending 2.1.3 zips could be a useful compromise. (one change was to make jed-checker happy so JED should accept them now)
